README: clarify Idris 1 vs 2 situation, and the compilation guide

attila-lendvai opened the pull request:
this is the fruit of my experience as a newcomer to Idris, and trying to compile Idris 1 from source.

juhp commented:
I haven't verified all the details but this LGTM anyway.

melted commented on README.md:
```diff
@@ -26,9 +26,18 @@ More information about building Idris from source has been detailed in the [Inst
+worked on anymore.
+
+[Idris 2](https://github.com/idris-lang/Idris2) is an early preview of
+the next generation of Idris, implemented in Idris 1.
```
Idris 2 is implemented in Idris 2, it was bootstrapped from Idris2-boot, which is the one implemented in Idris 1

melted replied:
Also, it's probably less unstable than an "early preview" now. But I don't know how Edwin would characterize it.

attila-lendvai commented:
ok, i've addressed the concerns by not saying anything about the bootstrap of idris 2. i guess it's more complicated than being discussed in the readme of idris 1... :)
